internal/gatewayapi/route.go

```diff
@@ -846,24 +846,22 @@ func (t *Translator) processUDPRouteParentRefs(udpRoute *UDPRouteContext, resour
 			)
 			continue
 		}
-		if len(udpRoute.Spec.Rules[0].BackendRefs) != 1 {
-			parentRef.SetCondition(udpRoute,
-				gwapiv1.RouteConditionResolvedRefs,
-				metav1.ConditionFalse,
-				"InvalidBackend",
-				"One and only one backend is supported",
-			)
-			continue
-		}
 
-		backendRef := udpRoute.Spec.Rules[0].BackendRefs[0]
-		ds, _ := t.processDestination(backendRef, parentRef, udpRoute, resources)
-		// Skip further processing if route destination is not valid
-		if ds == nil || len(ds.Endpoints) == 0 {
+		valid := true
+		for _, backendRef := range udpRoute.Spec.Rules[0].BackendRefs {
+			ds, _ := t.processDestination(backendRef, parentRef, udpRoute, resources)
+			// Skip further processing if route destination is not valid
```

Comment:
imo we should follow the same approach we use in HTTP case, where for an invalid destination, we track the weight as invalid, and build a weighted cluster based on valid and invalid weights across the multiple backendRefs

Reply:
May I know what's the benifit for doing that?

Reply:
its defined in the spec https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1alpha2.TCPRouteRule

Reply:
seems different to https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPBackendRef?

Reply:
also mentioned in https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteRule